stdlib/public/core/Identifiable.swift

```diff
@@ -10,11 +10,33 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// A class of types whose instances hold the value of an entity with stable identity.
+/// A class of types whose instances hold the value of an entity with stable
+/// identity.
+///
+/// `Identifiable` leaves the duration and scope of the identity unspecified.
+/// For example, identities could be:
+///
+/// - Guaranteed always unique (e.g. UUIDs).
+/// - Persistently unique per environment (e.g. database record keys).
+/// - Unique for the lifetime of a process (e.g. global incrementing integers).
+/// - Unique for the lifetime of an object (e.g. object identifiers).
+/// - Unique within the current collection (e.g. collection index).
+///
+/// It is up to both the conformer and the receiver of the protocol to document
+/// the nature of the identity.
+///
+/// Conforming to the Identifiable Protocol
+/// =======================================
+///
+/// `Identifiable` provides a default implementation for class types (using
+/// `ObjectIdentifier`), which is only guaranteed to remain unique for the
+/// lifetime of an object. If an object has a stronger notion of identity, it
+/// may be appropriate to provide a custom implementation.
 @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
 public protocol Identifiable {
 
-  /// A type representing the stable identity of the entity associated with `self`.
+  /// A type representing the stable identity of the entity associated with
+  /// `self`.
```

Comment:
We've moved away from using "`self`" in documentation this way, because for the reader, `self` can mean a variety of different things. I usually say "this instance" or something more specific.

```suggestion
  /// an instance.
```